podcasts/EpisodeLoadingController.swift

```diff
@@ -0,0 +1,138 @@
+                episodeLoadingModel.error = true
+                return
+            }
+
+            doneLoading()
+        }
+    }
+
+    func loadEpisode() async -> Bool {
+        await withCheckedContinuation { continuation in
+            // If we're missing the podcast, then load that and the episode
+            if self.podcast == nil {
+                ServerPodcastManager.shared.addMissingPodcastAndEpisode(episodeUuid: episodeUuid, podcastUuid: podcastUuid)
+            }
+            // If we're missing just the episode then get that
+            else {
+                _ = ServerPodcastManager.shared.addMissingEpisode(episodeUuid: episodeUuid, podcastUuid: podcastUuid)
+            }
+
+            // Verify they were added
+            let success = podcast != nil && episode != nil
+            continuation.resume(with: .success(success))
+        }
+    }
+
+    @MainActor func doneLoading() {
+        // Push to the controller and fade into it
+        let controller = EpisodeDetailViewController(episodeUuid: episodeUuid, source: .homeScreenWidget)
+
+        navigationController?.delegate = self
+        navigationController?.setViewControllers([controller], animated: true)
+    }
+
+    @MainActor required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// This is a bit of a hack to make it appear like the loading view is an overlay of the episode controller,
+/// but in reality it's in a navigation controller and we're pushing to it
+extension EpisodeLoadingController: UIViewControllerAnimatedTransitioning, UINavigationControllerDelegate {
+    func navigationController(_ navigationController: UINavigationController, animationControllerFor operation: UINavigationController.Operation, from fromVC: UIViewController, to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        self
+    }
+
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return 0.2
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        let fromView = transitionContext.viewController(forKey: UITransitionContextViewControllerKey.from)
+        let toView = transitionContext.viewController(forKey: UITransitionContextViewControllerKey.to)
+
+        guard let fromView, let toView else {
+            return
+        }
+
+        // Add the episode controller below the loading view so we can fade into it
+        transitionContext.containerView.insertSubview(toView.view, belowSubview: fromView.view)
+
+        let duration = transitionDuration(using: transitionContext)
+
+        // We delay the alpha transition here because the episode controller has a weird animation effect when it first loads
+        // 1 second is probably too long of a delay
+        UIView.animate(withDuration: duration, delay: 1.0) {
```

Comment:
I did some testing and 0.3 seconds seems to work well. 

```suggestion
        UIView.animate(withDuration: duration, delay: 0.3) {
```